packages/renderer/src/lib/onboarding/telemetry.ts

```diff
@@ -0,0 +1,133 @@
+export interface TelemetryStep {
+  id: string;
+  title: string;
+  durationMs: number;
+  error: undefined | string;
+}
+
+export interface TelemetryData {
+  extension: string;
+  skipped: boolean;
+  steps: TelemetryStep[];
+  stepsDetails?: any;
+  durationMs: number;
+  skipAtStep?: string;
+  errors?: string[];
+}
+
+export class TelemetrySession {
+  private data: TelemetryData;
+  private onboardingStartTime: number;
+  private stepStartTime: number;
+  private previousStepIndex: number;
+  private previousStepId: string;
+
+  constructor() {
+    this.data = {
+      extension: '',
+      skipped: false,
+      steps: [],
+      durationMs: 0,
+    };
+    this.onboardingStartTime = performance.now();
+    this.stepStartTime = performance.now();
+    this.previousStepIndex = -1;
+    this.previousStepId = '';
+  }
+
+  restart() {
+    this.onboardingStartTime = performance.now();
+    this.stepStartTime = performance.now();
+    this.previousStepIndex = -1;
+  }
+
+  startStep(i: number, id: string, title: string) {
+    if (this.previousStepIndex >= 0) {
+      this.savePreviousDuration();
+    }
+    this.previousStepIndex = i;
+    this.previousStepId = id;
+    this.data = telemetryAddStep(this.data, i, id, title);
+  }
+
+  setStepError(i: number, id: string, error: Error) {
+    this.data = telemetrySetStepError(this.data, i, id, error);
+  }
+
+  send(extensionName: string, skipped: boolean) {
+    this.savePreviousDuration();
+    this.data.extension = extensionName;
+    this.data.skipped = skipped;
+    this.data.durationMs = Math.round(performance.now() - this.onboardingStartTime);
+    window.telemetryTrack('onboarding', telemetryToSend(this.data));
+  }
+
+  private savePreviousDuration() {
+    const endTime = performance.now();
+    try {
+      const duration = Math.round(endTime - this.stepStartTime);
+      this.data = telemetrySetStepDuration(this.data, this.previousStepIndex, this.previousStepId, duration);
+    } catch {
+      // should not happen, as we should have added the step before
+      console.error('no step in telemetry');
+    }
+    this.stepStartTime = endTime;
+  }
+}
+
+function telemetryAddStep(data: TelemetryData, i: number, id: string, title: string): TelemetryData {
+  const completeId = getStepCompleteId(i, id);
+  data.steps.push({ id: completeId, title, durationMs: 0, error: undefined });
+  return data;
+}
+
+function telemetrySetStepDuration(data: TelemetryData, i: number, stepId: string, ms: number): TelemetryData {
+  const completeId = getStepCompleteId(i, stepId);
+  console.log('step duration', completeId, ms);
```

Comment:
```suggestion
  console.log('step duration', completeId, ms);
```

probably remove the debug console output